### ISSUE TITLE
unable to compile dither c++11 extension on gcc 4.6

### ISSUE THREAD

**Issue by jesg** [jesg/dither#5]
example:

``` text
$ rake compile
mkdir -p tmp/x86_64-linux/dither/2.1.0
cd tmp/x86_64-linux/dither/2.1.0
/home/jason/.rbenv/versions/2.1.0/bin/ruby -I. ../../../../ext/dither/extconf.rb
creating Makefile
cd -
cd tmp/x86_64-linux/dither/2.1.0
make
compiling ../../../../ext/dither/dither.cc
cc1plus: error: unrecognized command line option ‘-std=c++11’
Makefile:204: recipe for target 'dither.o' failed
make: *** [dither.o] Error 1
rake aborted!
Command failed with status (2): [make...]

Tasks: TOP => compile => compile:x86_64-linux => compile:dither:x86_64-linux => copy:dither:x86_64-linux:2.1.0 => tmp/x86_64-linux/dither/2.1.0/dither.so
(See full trace by running task with --trace)
```

**Comment by jesg** [comment]
gcc 4.6 does not support the c++11 option.  we need to replace this with c++0x.

**Comment by jesg** [comment]
still broken on gcc 4.6 after fixing the -std option.
```text
[jason@arch dither]$ rake compile
mkdir -p tmp/x86_64-linux/dither/2.1.0
cd tmp/x86_64-linux/dither/2.1.0
/home/jason/.rbenv/versions/2.1.0/bin/ruby -I. ../../../../ext/dither/extconf.rb
creating Makefile
cd -
cd tmp/x86_64-linux/dither/2.1.0
make
compiling ../../../../ext/dither/dither.cc
compiling ../../../../ext/dither/simple_constraint_handler.cc
In file included from /usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/algorithm:63:0,
                 from ../../../../ext/dither/simple_constraint_handler.h:15,
                 from ../../../../ext/dither/simple_constraint_handler.cc:10:
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h: In function ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, const _Tp&, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::vector<std::pair<long unsigned int, signed char> >*, std::vector<std::vector<std::pair<long unsigned int, signed char> > > >, _Tp = std::vector<std::pair<long unsigned int, signed char> >, _Compare = dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>]’:
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2266:78:   instantiated from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::vector<std::pair<long unsigned int, signed char> >*, std::vector<std::vector<std::pair<long unsigned int, signed char> > > >, _Compare = dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>]’
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2307:62:   instantiated from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::vector<std::pair<long unsigned int, signed char> >*, std::vector<std::vector<std::pair<long unsigned int, signed char> > > >, _Size = long int, _Compare = dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>]’
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:5446:4:   instantiated from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::vector<std::pair<long unsigned int, signed char> >*, std::vector<std::vector<std::pair<long unsigned int, signed char> > > >, _Compare = dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>]’
../../../../ext/dither/simple_constraint_handler.cc:32:14:   instantiated from here
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2234:4: error: no match for call to ‘(dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>) (std::vector<std::pair<long unsigned int, signed char> >&, const std::vector<std::pair<long unsigned int, signed char> >&)’
../../../../ext/dither/simple_constraint_handler.cc:29:14: note: candidates are:
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2234:4: note: bool (*)(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&) <conversion>
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2234:4: note:   candidate expects 3 arguments, 3 provided
../../../../ext/dither/simple_constraint_handler.cc:30:60: note: dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>
../../../../ext/dither/simple_constraint_handler.cc:30:60: note:   no known conversion for argument 2 from ‘const std::vector<std::pair<long unsigned int, signed char> >’ to ‘std::vector<std::pair<long unsigned int, signed char> >&’
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2237:4: error: no match for call to ‘(dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>) (const std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)’
../../../../ext/dither/simple_constraint_handler.cc:29:14: note: candidates are:
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2237:4: note: bool (*)(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&) <conversion>
/usr/lib/gcc/x86_64-unknown-linux-gnu/4.6.4/include/c++/bits/stl_algo.h:2237:4: note:   candidate expects 3 arguments, 3 provided
../../../../ext/dither/simple_constraint_handler.cc:30:60: note: dither::SimpleConstraintHandler::SimpleConstraintHandler(std::vector<signed char>&, std::vector<std::vector<signed char> >&)::<lambda(std::vector<std::pair<long unsigned int, signed char> >&, std::vector<std::pair<long unsigned int, signed char> >&)>
../../../../ext/dither/simple_constraint_handler.cc:30:60: note:   no known conversion for argument 1 from ‘const std::vector<std::pair<long unsigned int, signed char> >’ to ‘std::vector<std::pair<long unsigned int, signed char> >&’
Makefile:204: recipe for target 'simple_constraint_handler.o' failed
make: *** [simple_constraint_handler.o] Error 1
rake aborted!
Command failed with status (2): [make...]

Tasks: TOP => compile => compile:x86_64-linux => compile:dither:x86_64-linux => copy:dither:x86_64-linux:2.1.0 => tmp/x86_64-linux/dither/2.1.0/dither.so
```

**Comment by jesg** [comment]
fixed in 4e5a26f771b5a1b9bd6ed61929098b4813b9c946
